### PR TITLE
FE: Showcase S3 & Cloud Security Additional Steps

### LINKED ISSUE #886: Showcase Log Onboarding Steps
### Describe the ideal solution

We should separate the Step 1 and Step 2 of the log onboarding so users are reminded that they have to configure bucket notifications. The stack launched does not take care of it.

The flow should be something like this:

1. User adds basic information
2. User launches stack
3. We verify the connection and give a reminder to configure their S3 event notifications

### Describe your use cases

Easier onboarding - Updating S3 buckets is a non-trivial task that sometimes requires manual steps or code changes outside of Panther to work.

### Designs

https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/70fbde80-1195-4df9-a583-6962f57e3e19/commits/37590523135a8bd8a3e8a0ae7f6d46c8bb52b7ee/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/470D23BE-968A-466D-8F4F-12EF02FCB18E

### Acceptance Criteria

- After the user has setup IAM we do a health check (as it already happens). If the user fails it, we should show an error screen. If the user passes the check (which happens while you see the hourglass), then the user should go to the next step (instead of showing a success screen as it happens now).
- Add another step in the wizard right after verification step (inbetween the IAM setup and the final step) with the designs shown in the link above
- This step should only be present when onboarding an S3 source, not when editing it
- The "read more" should link to `LOG_ONBOARDING_SNS_DOC_URL` variable and you should  remove any other link where this constant is referenced **within the same wizard**
- When the user clicks the "I have setup ...." you should direct him to a success page (there's no API request or response involved)

### LINKED ISSUE #1482: Cloud Security doesn't find newly created S3 bucket
### Describe the bug

New resource doesn't appear in Cloud Security view and the reason is that if users don't add S3 notifications, we don't showcase the s3 bucket at all under the resources tab.

### Steps to reproduce

1. Go to Cloud Security -> Sources
2. Add a new account. Make sure to check the Real-time events
3. Go to your account. Create a new S3 bucket.
4. Search for the bucket, it doesn't appear

### Expected behavior

I expected to be able to find the S3 bucket once I created it. I waited for 10' minutes, but nothing came through. 

### Designs

https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/70fbde80-1195-4df9-a583-6962f57e3e19/commits/37590523135a8bd8a3e8a0ae7f6d46c8bb52b7ee/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/470D23BE-968A-466D-8F4F-12EF02FCB18E

### Acceptance Criteria

- After the user has setup IAM we do a health check (as it already happens). If the user fails it, we should show an error screen. If the user passes the check (which happens while you see the hourglass), then the user should see a different screen (instead of showing a success screen as it happens now).
- Add another screeen in the wizard right after verification with the designs shown in the link above
- This step should only be present when:
  -  onboarding an S3 source if `enable real-time` checkbox is checked
  - when editing it if the user has gone fro "disabled" real-time to enabled

Contrary to the designs:
- The "read more" should link to `CLOUD_SECURITY_REAL_TIME_DOC_URL` and you should remove any other place where this link is referenced **within the same wizard**
- The title should change to "Enable Real-Time monitoring"
- The subtitle should change to "You can now follow the steps found here to enable support for real-time monitoring of your Panther account"
- The helper text should change to "Panther does not validate if your account has been configured for real-time cloud security analysis. Failing to do this, will not allow Panther to instantly notify you when something goes wrong"
- When the user clicks the "I have setup ...." you should direct him to a success page (there's no API request or response involved)

### PR DESCRIPTION
**Pull request by 3nvi**
## Background

When adding a new source integration for either Cloud Security or Log Analysis (via S3), there are additional steps that need to be done. These are things  "outside of Panther" and normally include deploying CFN, enabling SNS notifications on S3 buckets, et.c

These steps were - up until now - not prominent to the user. This PR makes sure to accent them and have the user manually confirm that he/she did those steps.

Closes #886 
Closes #1482 

## Designs

https://app.abstract.com/projects/558a8de6-7134-4c8b-91c5-62074bb1279b/branches/70fbde80-1195-4df9-a583-6962f57e3e19/commits/37590523135a8bd8a3e8a0ae7f6d46c8bb52b7ee/files/c2bbc1a8-4778-4cd6-93a5-4630d8868201/layers/470D23BE-968A-466D-8F4F-12EF02FCB18E?mode=build&selected=root-26D4B6D0-D053-4E27-8F9F-C286FB215783

## Screenshots

Screen Shot 2020-11-20 at 7 07 53 PM
Screen Shot 2020-11-20 at 7 08 31 PM

## Changes

- Modify ComplianceWizard's Validation step to include a check for real-time CW events (if real-time checkbox is checked)
- Modify S3LogSourceWizard's Validation step to include a check for sns notifications (only during creation)
- Update UI of Log analysis onboarding to match that of Enterprise
- Update tests

## Testing

- `npm run test`

### CODE REVIEW

File: web/src/components/wizards/ComplianceSourceWizard/SourceConfigurationPanel/SourceConfigurationPanel.tsx [hunk 1]
```diff
@@ -72,10 +72,7 @@ const SourceConfigurationPanel: React.FC = () => {
             />
             <FormHelperText id="cweEnabled-description" ml={45}>
               Configure Panther to monitor all AWS resource changes in real-time through CloudWatch
-              Events.{' '}
-              <Link external href={CLOUD_SECURITY_REAL_TIME_DOC_URL}>
-                Read more
-              </Link>
+              Events.
```
Comment (3nvi) [inline web/src/components/wizards/ComplianceSourceWizard/SourceConfigurationPanel/SourceConfigurationPanel.tsx:75]: moved to a wizard panel of its own

File: web/src/components/wizards/ComplianceSourceWizard/ValidationPanel/ValidationPanel.tsx [hunk 2]
```diff
@@ -17,22 +17,32 @@
...
+  >();
   const [errorMessage, setErrorMessage] = React.useState('');
-  const { goToPrevStep, reset: resetWizard, currentStepStatus, setCurrentStepStatus } = useWizardContext(); // prettier-ignore
-  const { initialValues, submitForm, resetForm } = useFormikContext<ComplianceSourceWizardValues>();
+  const [shouldShowRealTimeScreen, setRealTimeScreenVisibility] = React.useState(
+    // creating a source with Real-time enabled
+    (!initialValues.integrationId && values.cweEnabled) ||
+      // updating a source to enable real-time
+      (!initialValues.cweEnabled && values.cweEnabled)
+  );
```
Comment (3nvi) [inline web/src/components/wizards/ComplianceSourceWizard/ValidationPanel/ValidationPanel.tsx:45]: only show the real-time configuration screen on certain occasions

File: web/src/hooks/useHover.test.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (C) 2020 Panther Labs Inc
```
Comment (3nvi) [inline web/src/hooks/useHover.test.tsx:20]: c/p from enterprise

File: web/src/hooks/useHover.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/**
```
Comment (3nvi) [inline web/src/hooks/useHover.tsx:1]: c/p from enterprise

File: web/src/pages/LogSourceOnboarding/LogSourceCard/LogSourceCard.tsx [hunk 5]
```diff
@@ -1,25 +1,18 @@
...
- *
- * You should have received a copy of the GNU Affero General Public License
- * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ * Panther Enterprise is licensed under the terms of a commercial license available from
+ * Panther Labs Inc ("Panther Commercial License") by contacting contact@runpanther.com.
+ * All use, distribution, and/or modification of this software, whether commercial or non-commercial,
+ * falls under the Panther Commercial License to the extent it is permitted.
  */
 
 import * as React from 'react';
```
Comment (3nvi) [inline web/src/pages/LogSourceOnboarding/LogSourceCard/LogSourceCard.tsx:28]: c/p from enterprise & modified for the OSS use case

**Review by alexmylonas — APPROVED** [review]
LGTM! 💯 

Remove those extra licenses and it's good to go

File: web/src/hooks/useHover.test.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ */
+
+/**
+ * Copyright (C) 2020 Panther Labs Inc
+ *
+ * Panther Enterprise is licensed under the terms of a commercial license available from
+ * Panther Labs Inc ("Panther Commercial License") by contacting contact@runpanther.com.
+ * All use, distribution, and/or modification of this software, whether commercial or non-commercial,
+ * falls under the Panther Commercial License to the extent it is permitted.
+ */
```
[the remark this answers is not in the document]
Comment (3nvi) [reply web/src/hooks/useHover.test.tsx:26]: nice catch

**Pull request merged**